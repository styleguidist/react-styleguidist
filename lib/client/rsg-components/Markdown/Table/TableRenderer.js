import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var space = _ref.space;
  return {
    table: {
      marginTop: 0,
      marginBottom: space[2],
      borderCollapse: 'collapse'
    }
  };
};

export var TableRenderer = function TableRenderer(_ref2) {
  var classes = _ref2.classes,
      children = _ref2.children;
  return /*#__PURE__*/React.createElement("table", {
    className: classes.table
  }, children);
};
TableRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  children: PropTypes.node.isRequired
};
export default Styled(styles)(TableRenderer);