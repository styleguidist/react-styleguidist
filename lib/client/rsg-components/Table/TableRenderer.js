import "core-js/modules/es.array.map";
import "core-js/modules/es.string.small";
import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
export var styles = function styles(_ref) {
  var space = _ref.space,
      color = _ref.color,
      fontFamily = _ref.fontFamily,
      fontSize = _ref.fontSize;
  return {
    table: {
      width: '100%',
      borderCollapse: 'collapse',
      marginBottom: space[4]
    },
    tableHead: {
      borderBottom: [[1, color.border, 'solid']]
    },
    cellHeading: {
      color: color.base,
      paddingRight: space[2],
      paddingBottom: space[1],
      textAlign: 'left',
      fontFamily: fontFamily.base,
      fontWeight: 'bold',
      fontSize: fontSize.small,
      whiteSpace: 'nowrap'
    },
    cell: {
      color: color.base,
      paddingRight: space[2],
      paddingTop: space[1],
      paddingBottom: space[1],
      verticalAlign: 'top',
      fontFamily: fontFamily.base,
      fontSize: fontSize.small,
      '&:last-child': {
        isolate: false,
        width: '99%',
        paddingRight: 0
      },
      '& p:last-child': {
        isolate: false,
        marginBottom: 0
      }
    }
  };
};
export var TableRenderer = function TableRenderer(_ref2) {
  var classes = _ref2.classes,
      columns = _ref2.columns,
      rows = _ref2.rows,
      getRowKey = _ref2.getRowKey;
  return /*#__PURE__*/React.createElement("table", {
    className: classes.table
  }, /*#__PURE__*/React.createElement("thead", {
    className: classes.tableHead
  }, /*#__PURE__*/React.createElement("tr", null, columns.map(function (_ref3) {
    var caption = _ref3.caption;
    return /*#__PURE__*/React.createElement("th", {
      key: caption,
      className: classes.cellHeading
    }, caption);
  }))), /*#__PURE__*/React.createElement("tbody", null, rows.map(function (row) {
    return /*#__PURE__*/React.createElement("tr", {
      key: getRowKey(row)
    }, columns.map(function (_ref4, index) {
      var render = _ref4.render;
      return /*#__PURE__*/React.createElement("td", {
        key: index,
        className: classes.cell
      }, render(row));
    }));
  })));
};
TableRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  columns: PropTypes.arrayOf(PropTypes.shape({
    caption: PropTypes.string.isRequired,
    render: PropTypes.func.isRequired
  }).isRequired).isRequired,
  rows: PropTypes.arrayOf(PropTypes.object).isRequired,
  getRowKey: PropTypes.func.isRequired
};
export default Styled(styles)(TableRenderer);