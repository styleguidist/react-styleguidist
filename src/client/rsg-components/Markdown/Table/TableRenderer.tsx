import React from 'react';
import PropTypes from 'prop-types';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../../typings';

const styles = ({ space }: Rsg.Theme) => ({
	table: {
		marginTop: 0,
		marginBottom: space[2],
		borderCollapse: 'collapse',
	},
});

interface TableProps extends JssInjectedProps {
	children: React.ReactNode;
}

export const TableRenderer: React.FunctionComponent<TableProps> = ({ classes, children }) => {
	return <table className={classes.table}>{children}</table>;
};

TableRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	children: PropTypes.any.isRequired,
};

export default Styled<TableProps>(styles)(TableRenderer);
