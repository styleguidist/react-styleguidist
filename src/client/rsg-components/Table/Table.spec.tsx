import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { TableRenderer, styles } from './TableRenderer';

const columns = [
	{
		caption: 'Name',
		// eslint-disable-next-line react/prop-types
		render: ({ name }: { name: string }) => <span>name: {name}</span>,
	},
	{
		caption: 'Type',
		// eslint-disable-next-line react/prop-types
		render: ({ type }: { type: string }) => <span>type: {type}</span>,
	},
];
const rows = [
	{ name: 'Quattro formaggi', type: 'pizza' },
	{ name: 'Tiramisu', type: 'desert' },
	{ name: 'Unicorn', type: 'animal' },
];
const props = {
	classes: classes(styles),
	getRowKey: (row: { name: string }) => row.name,
};

it('should render a table', () => {
	const renderer = createRenderer();
	renderer.render(<TableRenderer {...props} columns={columns} rows={rows} />);

	expect(renderer.getRenderOutput()).toMatchSnapshot();
});
