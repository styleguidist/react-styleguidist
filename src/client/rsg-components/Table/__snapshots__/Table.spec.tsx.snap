// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a table 1`] = `
<table
  className="table"
>
  <thead
    className="tableHead"
  >
    <tr>
      <th
        className="cellHeading"
      >
        Name
      </th>
      <th
        className="cellHeading"
      >
        Type
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="cell"
      >
        <span>
          name: 
          Quattro formaggi
        </span>
      </td>
      <td
        className="cell"
      >
        <span>
          type: 
          pizza
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="cell"
      >
        <span>
          name: 
          Tiramisu
        </span>
      </td>
      <td
        className="cell"
      >
        <span>
          type: 
          desert
        </span>
      </td>
    </tr>
    <tr>
      <td
        className="cell"
      >
        <span>
          name: 
          Unicorn
        </span>
      </td>
      <td
        className="cell"
      >
        <span>
          type: 
          animal
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
